### PR TITLE
[db migrator] Introduce the DB migration infrastructure

### PR DESCRIPTION
**Pull request by yxieca**
**- What I did**

- Add infrastructure to migrate DB contents from a lower version schema
  to a higher version schema, eventually reach latest version.
- if config load_minigraph is executed, the db schema version is always
  the latest.

**- How to verify it**
Some manual tests.

### CODE REVIEW

**Comment by yxieca** [comment]
config reload should also call db_migrator.py -o migtate

**Comment by yxieca** [comment]
retest this please

File: scripts/db_migrator.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+    def get_version(self):
+        version = self.configDB.get_entry(self.TABLE_NAME, self.TABLE_KEY)
+        if version and version[self.TABLE_FIELD]:
+            return version[self.TABLE_FIELD]
+
+        return 'version_unknown'
+
+
+    def set_version(self, version=CURRENT_VERSION):
```
Comment (yxieca) [inline scripts/db_migrator.py]: Cannot use self.CURRENT_VERSION here. Define as None and set to self.CURRENT_VERSION inside function when None.

File: scripts/db_migrator.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        self.TABLE_NAME      = 'VERSIONS'
+        self.TABLE_KEY       = 'DATABASE'
+        self.TABLE_FIELD     = 'VERSION'
+        self.configDB        = ConfigDBConnector()
+        self.configDB.db_connect('CONFIG_DB')
+
+
+    def migrate_pfc_wd_table(self):
+        # Migrate all data entries from table PFC_WD_TABLE to PFC_WD
+        data = self.configDB.get_talbe('PFC_WD_TABLE')
```
Comment (yxieca) [inline scripts/db_migrator.py]: s/talbe/table/

**Comment by prsunny** [comment]
retest this please

**Comment by yxieca** [comment]
retest this please

**Comment by lguohan** [comment]
retest this please

**Comment by lguohan** [comment]
retest this please

**Comment by yxieca** [comment]
retest this please

**Comment by yxieca** [comment]
Tested this change (uncommented the pfcwd upgrade), warm reboot from an image before to after worked.

**Comment by yxieca** [comment]
retest this please

**Comment by yxieca** [comment]
retest this please

**Comment by yxieca** [comment]
retest this please

**Pull request merged**